libflux/src/flux/semantic/nodes.rs

```diff
@@ -837,18 +837,11 @@ impl FunctionExpr {
     }
 }
 
-// Block represents a function block and is equivalent to a let-expression
```

Comment:
I think this comment is still accurate.  It might be good to add "A function block is an expression that evaluates to the argument of its terminating `ReturnStmt`."